src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -780,17 +783,47 @@ internal void Assign(IEnumerable<TopicPartitionOffset> partitions)
                 }
             }
 
-            ErrorCode err = Librdkafka.assign(handle, list);
+            ErrorCode err = ErrorCode.NoError;
+            Error error = null;
+
+            if (assignMethodErr != null) { err = assignMethodErr(handle, list); }
+            if (assignMethodError != null) { error = new Error(assignMethodError(handle, list)); }
```

Comment:
else if ?

Reply:
i've changed it but it's superfluous because execution can't get here if the first condition is true (exception thrown).

Reply:
this logic is actually incorrect. if assignMethodError returns IntPtr.Zero, this should set error to null, not an Error object. i have no idea how the tests were working - i'm quite sure they were ...